slasher/detection/service.go

```diff
@@ -108,21 +111,36 @@ func (ds *Service) detectHistoricalChainData(ctx context.Context) {
 	// slasher DB up to the current beacon node's head epoch we retrieved via gRPC.
 	// If no data was persisted from previous sessions, we request data starting from
 	// the genesis epoch.
-	for i := latestStoredEpoch; i < currentChainHead.HeadEpoch; i++ {
-		indexedAtts, err := ds.beaconClient.RequestHistoricalAttestations(ctx, i /* epoch */)
+	for epoch := latestStoredEpoch; epoch < currentChainHead.HeadEpoch; epoch++ {
+		indexedAtts, err := ds.beaconClient.RequestHistoricalAttestations(ctx, epoch)
 		if err != nil {
-			log.WithError(err).Errorf("Could not fetch attestations for epoch: %d", i)
+			log.WithError(err).Errorf("Could not fetch attestations for epoch: %d", epoch)
 		}
 		log.Debugf(
 			"Running slashing detection on %d attestations in epoch %d...",
 			len(indexedAtts),
-			i,
+			epoch,
 		)
-		// TODO(#4836): Run detection function for attester double voting.
-		// TODO(#4836): Run detection function for attester surround voting.
+		for _, att := range indexedAtts {
+			slashings, err := ds.detectAttesterSlashings(ctx, att)
+			if err != nil {
+				log.WithError(err).Error("Could not detect attester slashings")
+				continue
+			}
+			ds.submitAttesterSlashings(ctx, slashings)
```

Comment:
Should we only submit after processing all the indexedAtts? Instead of submitting each time we find a slashing?

Reply:
Thats not correct -  in the code above, we detect on the attestation and as soon as we find it, call submitAttesterSlashings